src/python/pants/backend/go/subsystems/golang.py

```diff
@@ -93,6 +100,96 @@ class GolangSubsystem(Subsystem):
         advanced=True,
     )
 
+    cgo_enabled = BoolOption(
+        "--cgo-enabled",
+        default=False,
+        help=softwrap(
+            """\
+            Enable Cgo, which allows calling C code from Go.
```

Comment:
Is it worth clarifying this is for both any third-party packages & also first-party? Maybe a paragraph like this

> Note that even if you don't use Cgo for first-party code, you may need to activate this for third-party packages.

Reply:
I plan to add some fields to the Go targets to control cgo support in a more refined manner. https://github.com/pantsbuild/pants/issues/16833